crypto/provider_core.c

```diff
@@ -442,16 +442,18 @@ static OSSL_PROVIDER *provider_new(const char *name,
 
     if ((prov = OPENSSL_zalloc(sizeof(*prov))) == NULL)
         return NULL;
+    if (!CRYPTO_NEW_REF(&prov->refcnt, 1)) {
+        ossl_provider_free(prov);
```

Comment:
I think you want OPENSSL_free(prov) here, ossl_provider_free will try to use the refcnt that doesn't exist.